mypy/test/teststubgen.py

```diff
@@ -310,3 +462,114 @@ def test(self, arg0: str) -> None:
+        def test(arg0: str) -> None:
+            """
+            test(arg0: argparse.Action)
+            """
+            pass
+        output = []  # type: List[str]
+        imports = []  # type: List[str]
+        mod = ModuleType('argparse', '')
+        generate_c_function_stub(mod, 'test', test, output, imports)
+        assert_equal(output, [
+            'def test(arg0: Action) -> Any: ...'
+        ])
+        assert_equal(imports, [])
+
+    def test_generate_c_function_other_module_ret(self) -> None:
+        """
+        Test that if return type references type from other module, module will be imported
+        """
+        def test(arg0: str) -> None:
+            """
+            test(arg0: str) -> argparse.Action
+            """
+            pass
+        output = []  # type: List[str]
+        imports = []  # type: List[str]
+        mod = ModuleType(self.__module__, '')
+        generate_c_function_stub(mod, 'test', test, output, imports)
+        assert_equal(output, [
+            'def test(arg0: str) -> argparse.Action: ...'
+        ])
+        assert_equal(imports, [
+            'import argparse'
+        ])
+
+    def test_generate_c_function_same_module_ret(self) -> None:
+        """
+        Test that if return type references type from same module but using full path, no module
+        will be imported, and type specification will be striped to local reference
+        """
+        def test(arg0: str) -> None:
+            """
+            test(arg0: str) -> argparse.Action
+            """
+            pass
+        output = []  # type: List[str]
+        imports = []  # type: List[str]
+        mod = ModuleType('argparse', '')
+        generate_c_function_stub(mod, 'test', test, output, imports)
+        assert_equal(output, [
+            'def test(arg0: str) -> Action: ...'
+        ])
+        assert_equal(imports, [])
+
+    def test_generate_c_type_with_overload_pybind11(self) -> None:
+        class TestClass:
+            def __init__(self, arg0: str) -> None:
+                """
+                __init__(*args, **kwargs)
+                Overloaded function.
+
+                1. __init__(self: TestClass, arg0: str) -> None
+
+                2. __init__(self: TestClass, arg0: str, arg1: str) -> None
+                """
+                pass
+        output = []  # type: List[str]
+        imports = []  # type: List[str]
+        mod = ModuleType(TestClass.__module__, '')
+        generate_c_function_stub(mod, '__init__', TestClass.__init__, output, imports,
+                                 self_var='self', class_name='TestClass')
+        assert_equal(output, [
+            '@overload',
+            'def __init__(*args, **kwargs) -> Any: ...',
+            '@overload',
+            'def __init__(self, arg0: str) -> None: ...',
+            '@overload',
+            'def __init__(self, arg0: str, arg1: str) -> None: ...',
+        ])
+        assert_equal(set(imports), {
```

Comment:
In generated stub will not be repeated, because when stub is generated (in `generate_stub_for_c_module`), it's made unique:
```
for line in sorted(set(imports)):
```

I could store it in set anyway, though it would introduce some unrelated changes.